catalyst-gateway/bin/src/service/common/types/cardano/nonce.rs

```diff
@@ -0,0 +1,125 @@
+//! Nonce
+
+use std::sync::LazyLock;
+
+use poem_openapi::{
+    registry::{MetaSchema, MetaSchemaRef},
+    types::{Example, ParseError, ParseFromJSON, ParseFromParameter, ParseResult, ToJSON, Type},
+};
+use serde_json::Value;
+
+use super::slot_no;
+
+/// Title.
+const TITLE: &str = "Nonce";
+/// Description.
+const DESCRIPTION: &str = "The current slot at the time a transaction was posted.
+Used to ensure out of order inclusion on-chain can be detected.
+
+*Note: Because a Nonce should never be greater than the slot of the transaction it is found in, 
+excessively large nonces are capped to the transactions slot number.*";
+/// Example.
+pub(crate) const EXAMPLE: u64 = slot_no::EXAMPLE;
+/// Minimum.
+const MINIMUM: u64 = 0;
+/// Maximum.
+const MAXIMUM: u64 = u64::MAX;
+
+/// Schema.
+#[allow(clippy::cast_precision_loss)]
+static SCHEMA: LazyLock<MetaSchema> = LazyLock::new(|| {
+    MetaSchema {
+        title: Some(TITLE.to_owned()),
+        description: Some(DESCRIPTION),
+        example: Some(EXAMPLE.into()),
+        maximum: Some(MAXIMUM as f64),
+        minimum: Some(MINIMUM as f64),
+        ..poem_openapi::registry::MetaSchema::ANY
+    }
+});
+
+/// Value of a Nonce.
+#[derive(Debug, Eq, PartialEq, Hash)]
+pub(crate) struct Nonce(u64);
+
+/// Is the Nonce valid?
```

Comment:
Should we have an actual logic instead of always returning true?
I guess we should have something like this:

```rust
value >=MINIMUM && value <=MAXIMUM
```

Reply:
It is that full range.  I accept that it is just returning true in reality, but i was following the same pattern where we are embedding the validation in a consistent way across these custom types.  The idea being, its more uniform from an implementation perspective, but the compiler should actually just eliminate all this code as its a constant true.

Reply:
Its probably clearer to use what you suggest.  The aim was to give a solid pattern for other devs to follow when implementing new types.